src/main/scala/org/ergoplatform/nodeView/state/ErgoState.scala

```diff
@@ -100,24 +105,44 @@ object ErgoState extends ScorexLogging {
           .map(in => checkBoxExistence(in.boxId))
           .sequence
 
-        lazy val boxes: Try[(List[ErgoBox], List[ErgoBox])] = dataBoxesTry.flatMap(db => boxesToSpendTry.map(bs => (db, bs)))
+        lazy val boxes: Try[(List[ErgoBox], List[ErgoBox])] = dataBoxesTry
+          .flatMap(db => boxesToSpendTry.map(bs => (db, bs)))
 
-        val vs = tx.validateStateless()
+        val res = tx.validateStateless()
           .validateNoFailure(txBoxesToSpend, boxesToSpendTry)
           .validateNoFailure(txDataBoxes, dataBoxesTry)
           .payload[Long](r.value)
-          .validateTry(boxes, e => ModifierValidator.fatal("Missed data boxes", e)) { case (_, (dataBoxes, toSpend)) =>
-            tx.validateStateful(toSpend.toIndexedSeq, dataBoxes.toIndexedSeq, currentStateContext, r.value)(verifier).result
-          }
-
-        execTx(tail, vs)
+          .validateTry(boxes, e => ModifierValidator.fatal("Missed data boxes", e)) {
```

Comment:
Most consensus-critical code must be simplified, not being made more complex. Please consider moving new logic to a separate decorator.